File: examples/extending-objects/Abstract Classes/Java/Person.java
public abstract class Person {
  protected String firstName;
  protected String lastName;
  protected int age;
  
  public Person(String firstName, String lastName, int age) {
    this.firstName = firstName;
    this.lastName = lastName;
    this.age = age;
  }
  
  public String toString() {
    return this.firstName + " " + this.lastName + ", " + this.age;
  }
  
  public abstract void increaseAge();
}

File: examples/extending-objects/Abstract Classes/Kotlin/Person.kt
// Assuming everything is in the same package

abstract class Person (firstName : String, lastName : String, age : Int) {
  val firstName : String
  val lastName : String
  var age : Int

  init {
	this.firstName = firstName
	this.lastName = lastName
	this.age = age   
  }
  
  abstract fun toString() : String {
    return this.firstName + " " + this.lastName + ", " + this.age;
  }
}